gateway-discovery-cm/src/main/java/org/apache/knox/gateway/topology/discovery/cm/ServiceModel.java

```diff
@@ -37,6 +37,9 @@
   private final String roleType;
   private final String serviceUrl;
 
+  // Metadata for the model object, which is not directly from the service or role configuration
+  private final Map<String, String> modelProperties = new ConcurrentHashMap<>();
```

Comment:
I'm wondering if `qualifyingServiceParameters` would be a better name for this field?

Reply:
You're probably correct. I went back and forth on that naming choice.

Reply:
Other than that it looks good to me. Will you submit a new PS?

Reply:
New PS in progress...